x/ratelimit/keeper/msg_server_test.go

```diff
@@ -41,19 +41,50 @@ func (s *KeeperTestSuite) TestMsgServer_AddRateLimit() {
 	denom := addRateLimitMsg.Denom
 	channelId := addRateLimitMsg.ChannelId
 
-	// TODO: Uncomment once function is implemented
-	_ = denom
-	_ = channelId
-	// // Add a rate limit successfully
-	// _, err := s.GetMsgServer().AddRateLimit(sdk.WrapSDKContext(s.Ctx), addRateLimitMsg)
-	// s.Require().NoError(err)
+	// Add a rate limit successfully
+	_, err := s.GetMsgServer().AddRateLimit(sdk.WrapSDKContext(s.Ctx), addRateLimitMsg)
+	s.Require().NoError(err)
 
-	// _, found := s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
-	// s.Require().True(found)
+	_, found := s.App.RatelimitKeeper.GetRateLimit(s.Ctx, denom, channelId)
+	s.Require().True(found)
+
+	// Check for duplicate rate limit
+	_, err = s.GetMsgServer().AddRateLimit(sdk.WrapSDKContext(s.Ctx), addRateLimitMsg)
+	s.Require().Error(err)
```

Comment:
can we check that this returns the KeyAlreadyExists error?

Reply:
ok